core/chains/evm/types/models.go

```diff
@@ -52,8 +55,23 @@ func NewHead(number *big.Int, blockHash common.Hash, parentHash common.Hash, tim
 	}
 }
 
+func (h *Head) BlockNumber() int64 {
+	return h.Number
+}
+
+func (h *Head) BlockHash() common.Hash {
+	return h.Hash
+}
+
+func (h *Head) GetParent() txmgrtypes.Head {
+	if h.Parent == nil {
+		return nil
+	}
+	return h.Parent
```

Comment:
we can probably simplify this?
```suggestion
	return h.Parent
```

Reply:
Actually, I'm not a golang expert, but I initially did what you suggested, but that causes crash in code.
Basically if we just return h.Parent, the receiving code doesn't treat it as nil, and crashes on referencing it!

Reply:
Yeah a typed-nil inside an interface is different than a pure nil.